agent/lib/kontena/workers/node_info_worker.rb

```diff
@@ -236,6 +253,54 @@ def calculate_container_time(container)
       0
     end
 
+    # @param [Array<Vmstat::Cpu>] prev_cpu_stats
+    # @param [Array<Vmstat::Cpu>] current_cpu_stats
+    # @return [Hash]
+    def calculate_average_cpu(prev_cpu_stats, current_cpu_stats)
+      all = prev_cpu_stats.zip(current_cpu_stats).map do |prev, current|
```

Comment:
I think this needs some comments to understand :) I guess the `all` will be shaped like:

```
[
  [cpu0_system_percentage, cpu0_user_percentage, cpu0_idle_percentage],
  ...
  [cpuN_system_percentage, cpuN_user_percentage, cpuN_idle_percentage],
]
```